hail_search/queries/mito.py

```diff
@@ -112,6 +118,215 @@ class MitoHailTableQuery(BaseHailTableQuery):
+                        False
+                    )) & (hl.is_defined(ht.wes_passes_inheritance[x[0]]) | hl.if_else(
+                        family_idx_map.get(x[1][0]['familyGuid'], hl.empty_dict(hl.tstr, hl.tint32)).contains(SAMPLE_TYPE_WGS),
+                        hl.is_defined(ht.wgs_passes_inheritance[family_idx_map[x[1][0]['familyGuid']][SAMPLE_TYPE_WGS]]),
+                        False
+                    )), x[1]
+            )),
+            wgs_family_entries=hl.enumerate(ht[WGS_FAMILY_ENTRIES_FIELD]).map(
+                lambda x: hl.or_missing(
+                    (hl.is_defined(ht.wgs_passes_quality[x[0]]) | hl.if_else(
+                        family_idx_map.get(x[1][0]['familyGuid'], hl.empty_dict(hl.tstr, hl.tint32)).contains(SAMPLE_TYPE_WES),
+                        hl.is_defined(ht.wes_passes_quality[family_idx_map[x[1][0]['familyGuid']][SAMPLE_TYPE_WES]]),
+                        False
+                    )) & ( hl.is_defined(ht.wgs_passes_inheritance[x[0]]) | hl.if_else(
+                        family_idx_map.get(x[1][0]['familyGuid'], hl.empty_dict(hl.tstr, hl.tint32)).contains(SAMPLE_TYPE_WES),
+                        hl.is_defined(ht.wes_passes_inheritance[family_idx_map[x[1][0]['familyGuid']][SAMPLE_TYPE_WES]]),
+                        False
+                    )), x[1]
+            ))
+        )
+        ht = ht.transmute(
+            family_entries=hl.coalesce(
+                ht.wes_family_entries, hl.empty_array(ht.wes_family_entries.dtype.element_type)).extend(
+                hl.coalesce(ht.wgs_family_entries, hl.empty_array(ht.wgs_family_entries.dtype.element_type))
+            ),
+            filters=hl.coalesce(ht.wes_filters, hl.empty_set(hl.tstr)).union(
+                hl.coalesce(ht.wgs_filters, hl.empty_set(hl.tstr))
+            )
+        )
+        return ht
+
+
+    def _import_and_filter_multiple_project_hts(
+        self, project_samples: dict, n_partitions=MAX_PARTITIONS, **kwargs
+    ) -> tuple[hl.Table, hl.Table]:
+        sample_types = set()
+        for sample_dict in project_samples.values():
+            sample_types.update(sample_dict.keys())
+        if len(sample_types) == 1:
+            return super()._import_and_filter_multiple_project_hts(project_samples, n_partitions, **kwargs)
+
+        self._has_both_sample_types = True
+        entries_hts_map = self._load_prefiltered_project_hts_both_sample_types(project_samples, **kwargs)
+        return self._filter_entries_ht_both_sample_types(entries_hts_map, n_partitions, **kwargs)
+
+
+    def _load_prefiltered_project_hts_both_sample_types(
+        self, project_samples: dict, n_partitions=MAX_PARTITIONS, **kwargs
+    ) -> dict[str, list[tuple[hl.Table, dict]]]:
+        entries_hts_map = {}
+        chunk_size = 64
+        project_guid_chunks: list[list[str]] = []
+        current_project_guid_chunk: list[str] = []
```

Comment:
I commented this above, but I do not understand why the logic for merging the project tables needs to be different depending on how many sample types there are. As long as the result of reading in the multi sample type tables ins a single filtered table for the project, the existing logic should just work